content/en/community/hackathons/sessions/baby-steps/content/Setup_Helloworld.md

```diff
@@ -0,0 +1,287 @@
+Since the `helloworld` application is fairly simple, it does not use any external libray, so we can leave the `libs/` directory empty.
+
+```bash
+$ mkdir .unikraft
+
+$ cd .unikraft/
+
+$ git clone https://github.com/unikraft/unikraft unikraft
+
+$ mkdir libs
+```
+
+### Configuration
+
+Now that we have our setup ready, we can head back to the `helloworld/` directory and start configuring our application!
+
+```bash
+$ cd ../
+
+$ make menuconfig
+```
+
+This will open a menu-driven user interface in which you can choose the target architecture, virtualization platform, and configuration options for both internal and external libraries of the resulting unikernel image.
+
+We are met with the following configuration menu. Let's pick the architecture:
+
+![arch selection menu](/community/hackathons/sessions/baby-steps/images/menuconfig_select_arch.png)
+
+![arch selection menu2](/community/hackathons/sessions/baby-steps/images/menuconfig_select_arch2.png)
+
+![arch selection menu3](/community/hackathons/sessions/baby-steps/images/menuconfig_select_arch3.png)
+
+Now, press `Exit` (or hit the `Esc` key twice) until you return to the initial menu.
+
+We have now set our desired architecture, let's now proceed with the platform.
+We will choose only the `KVM` platform (press the `Y` key to select it), which is supported by QEMU, the program we will use to launch our virtual machine after compilation:
+
+![plat selection menu](/community/hackathons/sessions/baby-steps/images/menuconfig_select_plat.png)
+
+![plat selection menu2](/community/hackathons/sessions/baby-steps/images/menuconfig_select_plat2.png)
+
+For the moment, this is sufficient. Press the `Save` button and exit the configuration menu by repeatedly selecting `Exit`.
+
+### Building
+We can now build our application. We can do this by simply invoking `make`:
+
+```bash
+$ make -j$(nproc)
+```
+
+This command invokes the build system by compiling the application on `$(nproc)` different threads, for faster build times.
+The compilation process includes all of the Unikraft core internal libraries selected by default for this app, along with the application source files (just `main.c` for `helloworld`).
+The process should end with the following output:
+
+```bash
+[...]
+  LD      helloworld_qemu-x86_64.dbg
+  UKBI    helloworld_qemu-x86_64.dbg.bootinfo
+  SCSTRIP helloworld_qemu-x86_64
+  GZ      helloworld_qemu-x86_64.gz
+make[1]: Leaving directory '/home/X/workdir/helloworld/.unikraft/unikraft'
+```
+### Running
+Now that the unikernel image has been sucessfully generated, we can finally run it using the QEMU Virtual Machine Monitor:
+
+```bash
+$ qemu-system-x86_64 -kernel build/helloworld_qemu-x86_64 -nographic
```

Comment:
I would explain the `qemu` arguments here. Just a short description of why we use `-nographic`, and that we specify the we pass the kernel image via the `-kernel` option.

Reply:
I would also add the `arm64` part for helloworld. Just mention that we need to change the architecture, the build and setup remains the same, and then give the `qemu` command. The point is for them to see the `qemu-system-aarch64` command and realize that the arguments are the same, with the extra `-machine` and `-cpu`. Do not spend time insisting on anything else but the running part.